### ISSUE TITLE
sc2034 false positive with declare -F

### ISSUE THREAD

**Issue by eparis** [koalaman/shellcheck#371]
```bash
#!/bin/bash

__custom_func() {
    echo "In Custom!"
}

declare -F __custom_func >/dev/null && __custom_func
```

Results in:
```
declare -F __custom_func >/dev/null && __custom_func
                 ^––SC2034 __custom_func appears unused. Verify it or export it.
```

**Comment by koalaman** [comment]
Fixed in 09b77884

**Comment by stdedos** [comment]
I cannot verify that in the mockup:
```
if declare -f -F ask > /dev/null ; then
	ask "Do you want to such and such?" Y && (bash) && printf "Done"
fi
```
